canonical-docs/phone-docs
# Python app and dependencies

Add a python app for serving the docs.

Fixes https://github.com/canonical-websites/docs.ubuntu.com/issues/99

QA
--


First build docs. This will take a *very* long time. Check back after 30 mins.

``` bash
documentation-builder --base-directory documents --output-path templates
```

(If you get impatient you can inspect its progress by comparing `find documents -name '*.md' | wc -l` with `find templates/en/ -name '*.html' | wc -l`)

Install dependencies

``` bash
python3 -m venv env
env/bin/pip install -r requirements.txt
```

Run the server

```
FLASK_APP=app.py env/bin/flask run
```

Now go to <http://127.0.0.1:5000> and check it looks like <https://docs.ubuntu.com/phone/>.
